File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    return dfs(0, numbers, target, 0)

def dfs(depth, numbers, target, total):
    if depth == len(numbers):
        return 1 if total == target else 0
    
    plus = dfs(depth + 1 ,numbers, target, total + numbers[depth])
    minus = dfs(depth + 1 ,numbers, target, total + (numbers[depth] * (-1))) 
    
    return plus + minus
    
    

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.Queue;

class Solution {
    int dfs(int[] numbers, int target, int depth, int current){
        if(depth == numbers.length){
            if(current == target) return 1;
            return 0;
        }
        return dfs(numbers, target, depth+1, current+numbers[depth]) + dfs(numbers, target, depth+1, current-numbers[depth]);
    }
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }
}